src/tools/pkcs11-tool.c

```diff
@@ -2863,11 +2863,12 @@ static int gen_keypair(CK_SLOT_ID slot, CK_SESSION_HANDLE session,
 				if (!find_mechanism(slot, CKF_GENERATE_KEY_PAIR, mtypes, mtypes_num, &opt_mechanism))
 					util_fatal("Generate RSA mechanism not supported");
 
-			if (size == NULL)
-				util_fatal("Unknown key pair type %s, expecting RSA:<nbytes>", type);
 			key_length = (unsigned long)atol(size);
-			if (key_length != 0)
+			if (key_length != 0) {
+				if (*size != '\0')
+					util_fatal("Unknown key pair type %s, expecting RSA:<nbytes>", type);
 				modulusBits = key_length;
+			}
```

Comment:
so in that case, this logic of this block is somehow flawed as it fails the tests setup now with an error like:
```
Setup SoftHSM
=======================================================
The token has been initialized and is reassigned to slot 148272278
Using slot 0 with a present token (0x8d67496)
error: Unknown key pair type RSA:1024, expecting RSA:<nbytes>
Aborting.
Couldn't generate RSA:1024 key pair
====> ERROR: Failed to set up card
```

Reply:
obviously it should have been `if (key_length == 0) {`

I fixed that and additionally removed the default 1024 for the reasons described above

Reply:
Thanks. This is resolved. Now just one more tests depends on the defaults:

https://github.com/OpenSC/OpenSC/blob/master/tests/test-pkcs11-tool-allowed-mechanisms.sh#L24

Please, fix also that one so we can merge this.

Reply:
Is there an option to view tests/test-suite.log from the CI pipeline?

Reply:
Its uploaded as part of artifacts (can be downloaded from the workflow summary):

https://github.com/OpenSC/OpenSC/blob/master/.github/workflows/linux.yml#L27

The other option that looks like it does not work now is dumping the logs into the github output (but with the set -e it will never be executed as the execution ends with the failed `make check`):

https://github.com/OpenSC/OpenSC/blob/c3545018d059b4debde33b9f34de719dd41e5531/.github/build.sh#L67
https://github.com/OpenSC/OpenSC/blob/master/.github/dump-logs.sh

If you would find it useful to see the logs in the output, I can try to get that working.

Reply:
fixed now.

The output of the logs in case of failure would be useful for future debugging, if you want to look at this again later.

Reply:
For dumping logs, I filled #3005.